File: 8.Sorting/Java/2_bubble_sort.java
class BubbleSort {
    void bblSort(int arr[]) {
        for (int i=arr.length-1; i>0; i--) {
            for (int j=arr.length-1; j>0; j--) {
                if (arr[j] < arr[j-1]) {
                    int temp = arr[j];
                    arr[j] = arr[j-1];
                    arr[j-1] = temp;
                }
            }
        }
    }
    void show(int arr[], String msg) {
        System.out.println(msg);
        for (int i=0; i<arr.length; i++) {
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }

    public static void main(String arg[]) {
        int arr[] = {10, 2, 22, 43, 9, 55, 0};
        BubbleSort o = new BubbleSort();
        o.show(arr, "before");
        o.bblSort(arr);
        o.show(arr, "after");
    }
}

File: 8.Sorting/C/2_bubble_sort.c
#include<stdio.h>

void show(int arr[], int size, char* msg) {
    printf("%s\n", msg);
    for (int i=0; i<size; i++) {
        printf("%s ", arr[i]);
    }
    printf("\n");
}

void bblSort(int arr[], int size) {
    for (int i=size-1; i>0; i--) {
        for (int j=size-1; i>0; j--) {
            if (arr[j] < arr[j-1]) {
                int temp = arr[i];
                arr[j-1] = arr[i];
                arr[i] = temp;
            }
        }
    }
}

int main() {
    int arr[] = {128, 1, 18, 9, 0, 11, 22};
    int size = sizeof(arr) / sizeof(int);
    show(arr, size, "before");
    bblSort(arr, size);
    show(arr, size, "after");
    
}